src/Engine/GameInstance.java

```diff
@@ -218,20 +222,43 @@ private String getSaveName()
     return sb.toString();
   }
   
+  public static boolean isSaveCompatible(String filename)
+  {
+    System.out.println(String.format("Checking compatibility of save %s", filename));
+
+    GameVersion verInfo = null;
+    boolean verMatch = false;
+    try (FileInputStream file = new FileInputStream(filename); ObjectInputStream in = new ObjectInputStream(file);)
+    {
+      verInfo = (GameVersion) in.readObject();
+      if( new GameVersion().isEqual(verInfo) )
+      {
+        verMatch = true;
+      }
+    }
+    catch (Exception ex)
+    {
+      System.out.println(ex.toString());
+    }
+
+    return verMatch;
+  }
+  
   public static GameInstance loadSave(String filename)
   {
     System.out.println(String.format("Deserializing game data from %s", filename));
     
+    GameVersion verInfo = null;
     GameInstance load = null;
-    try
+    try (FileInputStream file = new FileInputStream(filename); ObjectInputStream in = new ObjectInputStream(file);)
     {
-      FileInputStream file = new FileInputStream(filename);
-      ObjectInputStream in = new ObjectInputStream(file);
+      verInfo = (GameVersion) in.readObject();
+      if( !new GameVersion().isEqual(verInfo) )
+      {
+        throw new Exception(String.format("Deserializing failed! Save is incompatible version: %s", verInfo.toString()));
```

Comment:
Ah, you do print it here. However, we can't get here if the version doesn't match, because `isSaveCompatible` will prevent it from being added to the list.

Reply:
I think it might be better to check for save compatibility only when actually attempting to load it. It's going to cause confusion of a .svp just doesn't show up in the load menu, and there's no good indicator as to why.

We could even modify the display name in the menu (add a star/version number/whatever) so it's clear we see it, and that it won't work.

Reply:
Aight, I'll see about doing that.